Pointer icons for native macOS

Thomas-Vos opened the pull request:
Set the correct NSCursor depending on the pointer icon provided by compose.

## Testing

Tested using mpp demo sample on macOS arm64. Open a screen with a text field and the cursor changes.

Thomas-Vos committed:
Pointer icons for native macOS

igordmn commented on compose/ui/ui/src/macosMain/kotlin/androidx/compose/ui/window/ComposeWindow.macos.kt:
```diff
@@ -93,6 +95,14 @@ private class ComposeWindow(
             override val windowInfo get() = _windowInfo
             override val textInputService get() = macosTextInputService
+            override fun setPointerIcon(pointerIcon: PointerIcon) {
+                when (pointerIcon) {
```
We should set it to `NSCursor.arrowCursor.set()` in `else` case

Thomas-Vos replied:
Fixed, thanks!

Thomas-Vos committed:
fixup! Pointer icons for native macOS

Thomas-Vos committed:
fixup! Pointer icons for native macOS

Thomas-Vos commented:
@igordmn thanks for the review. Changed the PR a bit as I just found out there was already an implementation for darwin which I split into uikit and macos source sets.
